aodndata/soop/soop_xbt_nrt.py

```diff
@@ -1,13 +1,394 @@
+    template.variables['LONGITUDE']['_data'] = profile_data['profile_geotime']['longitude']
+
+    template.variables['TEMP']['_data'] = profile_data['profile_data']['temp'].values
+    template.variables['DEPTH']['_data'] = profile_data['profile_data']['depth'].values
+
+    template.variables['TEMP_quality_control']['_data'] = np.int8(profile_data['profile_data']['glob_gtspp_temp'])
+    template.variables['DEPTH_quality_control']['_data'] = np.int8(profile_data['profile_data']['glob_gtspp_depth'])
+
+    # issue with the aodntools NetCDF suite not updating some attributes to the correct type. The following does this
+    # manually
+    att_to_convert = ["flag_values", "valid_min", "valid_max"]
+    for att in att_to_convert:
+        for var in ['TEMP_quality_control', 'DEPTH_quality_control']:
+            template.variables[var][att] = np.int8(template.variables[var][att])
+
+    time_attr_format = '%Y-%m-%dT%H:%M:%SZ'
+    time_coverage_start = datetime.datetime.strftime(profile_data['profile_geotime']['date_utc'], time_attr_format)
+    time_coverage_end = time_coverage_start
+
+    abstract = get_info_xbt_config(profile_data['profile_metadata']['XBT_line'])['abstract']
+    title = get_info_xbt_config(profile_data['profile_metadata']['XBT_line'])['title']
+
+    template.global_attributes.update({
+        'abstract': abstract,
+        'title': title,
+        'time_coverage_start': time_coverage_start,
+        'time_coverage_end': time_coverage_end,
+        'featureType': 'profile',
+        'date_created': datetime.datetime.utcnow().strftime(time_attr_format),
+        'geospatial_lat_min': profile_data['profile_geotime']['latitude'],
+        'geospatial_lat_max': profile_data['profile_geotime']['latitude'],
+        'geospatial_lon_min': profile_data['profile_geotime']['longitude'],
+        'geospatial_lon_max': profile_data['profile_geotime']['longitude'],
+        'geospatial_vertical_min': profile_data['profile_data']['depth'].min(),
+        'geospatial_vertical_max': profile_data['profile_data']['depth'].max(),
+        'history': "File created {date_created}".format(
+            date_created=datetime.datetime.utcnow().strftime(time_attr_format))
+    })
+
+    nc_filename = 'IMOS_SOOP-XBT_T_{date_start}_{xbt_line}_FV00_ID_{imo_number}.nc'.format(
+        date_start=datetime.datetime.strftime(profile_data['profile_geotime']['date_utc'], '%Y%m%dT%H%M%SZ'),
+        xbt_line=profile_data['profile_metadata']['XBT_line'],
+        imo_number=profile_data['profile_metadata']['imo_number']
+    )
+
+    netcdf_path = os.path.join(output_dir, nc_filename)
+    template.to_netcdf(netcdf_path)
+    return netcdf_path
+
+
+def _config_parser(conf_file):
+    """ parse a config file """
+    parser = ConfigParser()
+    parser.optionxform = str  # to preserve case
+    conf_file_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), conf_file)
```

Comment:
You shouldn't try to go hunting around on the filesystem when code is part of a package, it's a recipe for pain. You've imported `from pkg_resources import resource_filename` above, so if you really need a custom config file which is part of the package, you should use that excursively to discover the path to the file.

In addition to that, non-Python files will only be packaged if they are listed in the setup.py in the `package_data` dict.

Rather than list individually however, it should be added to a `configuration` directory under aodndata, and we can then just include `configuration/*.ini` rather than editing the package_data list for every new file.

https://github.com/aodn/python-aodndata/blob/master/setup.py#L101

Reply:
done